instrumentation-core/servlet-2.2/src/main/java/io/opentelemetry/instrumentation/servlet/ServletHttpServerTracer.java

```diff
@@ -123,8 +123,16 @@ protected String requestHeader(HttpServletRequest httpServletRequest, String nam
     return httpServletRequest.getHeader(name);
   }
 
-  private static String getSpanName(HttpServletRequest request) {
+  public static String getSpanName(HttpServletRequest request) {
     String spanName = request.getServletPath();
+    String pathInfo = request.getPathInfo();
+    // getServletPath() returns "" when request doesn't target any servlet or
+    // when it targets servlet mapped to /*
+    // for example websphere liberty does not use a servlet for static content so
+    // requests for static files will have "" for servlet path
```

Comment:
It is called from LibertyUpdateSpanAdvice for all requests including the ones that don't have a servelt

Reply:
I don't understand. In what sense "is called"? `LibertyUpdateSpanAdvice` uses `LibertyHttpServerTracer`, not servlet tracer, doesn't it?

Reply:
`LibertyHttpServerTracer` extends `Servlet3HttpServerTracer`

Reply:
It can. On liberty there is no default servlet for static content, but we still call getSpanName for requests that are for static files. Similarly default servlet can be removed/disable on at least tomcat, in that case it is possible to have a filter that sees the request but no servlet.